Add new chart xychart (#4413)
mermaid-js/mermaid: packages/mermaid/src/docs/syntax/xyChart.md

> It's important to note that while the current implementation of mermaid-js includes these two chart types, the framework is designed to be dynamic and adaptable. Therefore, it has the capacity for expansion and the inclusion of additional chart types in the future. This means that users can expect an evolving suite of charting options within the XY chart module, catering to various data visualization needs as new chart types are introduced over time.

## Example

```mermaid-example
xychart-beta
    title "Sales Revenue"
    x-axis [jan, feb, mar, apr, may, jun, jul, aug, sep, oct, nov, dec]
    y-axis "Revenue (in $)" 4000 --> 11000
    bar [5000, 6000, 7500, 8200, 9500, 10500, 11000, 10200, 9200, 8500, 7000, 6000]
    line [5000, 6000, 7500, 8200, 9500, 10500, 11000, 10200, 9200, 8500, 7000, 6000]
```

## Syntax

```note

> all text values can be single word without ", if multiple line required we have to use ".

Author: I am not good at English, apologise for this, it makes sense what you said so changed the text as you suggested.

> all text values can be single word without ", if multiple line required we have to use ".

Reviewer: I'd like to clarify. What do we support without double quotes? **multi-line** values or or **single-word** values?

> all text values can be single word without ", if multiple line required we have to use ".

Author: You are absolutely right instead of multiple lines it would be multiple words.

```

### Orientations

> The chart can be drawn horizontal or vertical, default value is vertical

Author: Done

```
xychart-beta horizontal
...
```

### Title

The title is a short description of the chart and it will always render on top of the chart.

#### Example

```
xychart-beta

>     title "This is a sample example"

Author: Done

    ...
```

```note

> if the title single word no need to use ", but if it has space " is needed

Author: Yes as in the other diagram title is parsed differently so to make it clear I added this line.

> if the title single word no need to use ", but if it has space " is needed

Reviewer: Then let's fix it a little bit
```suggestion
If the title is a single word one no need to use `"`, but if it has space `"` is needed
```

> if the title single word no need to use ", but if it has space " is needed

Author: Done

```

### x-axis

The x-axis primarily serves as a categorical value, although it can also function as a numeric range value when needed.

#### Example

1. `x-axis title min --> max` x-axis will function as numeric with the given range
2. `x-axis "title with space" [cat1, "cat2 with space", cat3]` x-axis if categorical, categories are text type

### y-axis

> The y-axis is employed to represent numerical range values, it can't have categorical values.

Author: Done

#### Example

1. `y-axis title min --> max`
2. `y-axis title` it will only add the title, the range will be auto generated from data.

```note
Both x and y axis are optional if not provided we will try to create the range
```

### Line chart

A line chart offers the capability to graphically depict lines.

#### Example

1. `line [2.3, 45, .98, -3.4]` it can have all valid numeric values.

### Bar chart

A bar chart offers the capability to graphically depict bars.

#### Example

1. `bar [2.3, 45, .98, -3.4]` it can have all valid numeric values.

#### Simplest example

> Every grammer are optional other than the chart name and one data set, so you will be able to draw a chart will a simple config like

Reviewer: ```suggestion
The only two things required are the chart name (`xychart-beta`) and one data set. So you will be able to draw a chart with a simple config like
```

Author: Done

```
xychart-beta
    line [+1.3, .6, 2.4, -.34]
```

## Chart Configurations

| Parameter                | Description                                    | Default value |
| ------------------------ | ---------------------------------------------- | :-----------: |
| width                    | Width of the chart                             |      700      |
| height                   | Height of the chart                            |      500      |
| titlePadding             | Top and Bottom padding of the title            |      10       |
| titleFontSize            | Title font size                                |      20       |
| showTitle                | Title to be shown or not                       |     true      |
| xAxis                    | xAxis configuration                            |  AxisConfig   |
| yAxis                    | yAxis configuration                            |  AxisConfig   |

> | chartOrientation         | ('vertical' or 'horizontal')                   |  'vertical'   |

Reviewer: ```suggestion
| chartOrientation         | 'vertical' or 'horizontal'                     |  'vertical'   |
```

Author: Done

## Chart Theme Variables

```note
Themes for xychart resides inside xychart attribute so to set the variables use this syntax
%%{init: { "themeVariables": {"xyChart": {"titleColor": "#ff0000"} } }}%%
```

| Parameter        | Description                                             |
| ---------------- | ------------------------------------------------------- |
| backgroundColor  | Background color of the whole chart                     |
| titleColor       | Color of the Title text                                 |
| xAxisLableColor  | Color of the x-axis labels                              |
| xAxisTitleColor  | Color of the x-axis title                               |
| xAxisTickColor   | Color of the x-axis tick                                |
| xAxisLineColor   | Color of the x-axis line                                |
| yAxisLableColor  | Color of the y-axis labels                              |
| yAxisTitleColor  | Color of the y-axis title                               |
| yAxisTickColor   | Color of the y-axis tick                                |
| yAxisLineColor   | Color of the y-axis line                                |

> | plotColorPalette | String of colors seperated by comma eg "#f3456, #43445" |

Reviewer: ```suggestion
| plotColorPalette | String of colors separated by comma e.g. "#f3456, #43445" |
```

Author: Done
